tests/stdlib/map.c3

```diff
@@ -0,0 +1,73 @@
+@require_once "std/map.c3"
+@require_once "std/format.c3"
+@require_once "std/assert.c3"
+@require_once "std/string.c3"
+
+function main : () -> int = {
+    mut map : Map<int, StringView> = make<Map<int, StringView>>();
+
+    runtime_assert(map:empty() == true);
+    runtime_assert(map:size() == 0);
+
+    map:insert(0, sv("0"));
+    map:insert(1, sv("1"));
+    map:insert(2, sv("2"));
+    map:insert(3, sv("3"));
+
+    runtime_assert(map:contains(0));
+    runtime_assert(map:contains(1));
+    runtime_assert(map:contains(2));
+    runtime_assert(map:contains(3));
+
+    runtime_assert(map:empty() == false);
+    runtime_assert(map:size() == 4);
+
+    print(map:at(4));
+    print(map:at(3));
+    print(map:at(2));
+    print(map:at(1));
+    print(map:at(0));
+
+    for i in map:get_citer() {
+        printf(sv("% = %"), i.key, i.value);
+    }
+
+    runtime_assert(map:erase(0));
+    runtime_assert(!map:erase(4));
```

Comment:
Wanna change `!` to `not`. It'd match our `and`/ `or` syntax?

Reply:
I wouldn't be opposed to it. Should we make it non-overloadable then? To match the convention of `and`/`or`?

Also, it'd be pretty strange to change the meaning of the keyword `not`

Reply:
Yeah, idm that. It's not like an overload of `!` makes sense anyway